hw/mcu/stm/stm32l4xx/src/hal_nvreg.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <mcu/cortex_m4.h>
+#include <hal/hal_nvreg.h>
+#include <stm32l4xx_hal_pwr.h>
+
+/* There are 32 RTC backup registers in the stm32l4 family */
+#define HAL_NVREG_MAX (32)
+
+/* RTC backup registers are 32-bits wide */
+#define HAL_NVREG_WIDTH_BYTES (4)
+
+static volatile uint32_t *regs[HAL_NVREG_MAX] = {
+    &RTC->BKP0R,
+    &RTC->BKP1R,
+    &RTC->BKP2R,
+    &RTC->BKP3R,
+    &RTC->BKP4R,
+    &RTC->BKP5R,
+    &RTC->BKP6R,
+    &RTC->BKP7R,
+    &RTC->BKP8R,
+    &RTC->BKP9R,
+    &RTC->BKP10R,
+    &RTC->BKP11R,
+    &RTC->BKP12R,
+    &RTC->BKP13R,
+    &RTC->BKP14R,
+    &RTC->BKP15R,
+    &RTC->BKP16R,
+    &RTC->BKP17R,
+    &RTC->BKP18R,
+    &RTC->BKP19R,
+    &RTC->BKP20R,
+    &RTC->BKP21R,
+    &RTC->BKP22R,
+    &RTC->BKP23R,
+    &RTC->BKP24R,
+    &RTC->BKP25R,
+    &RTC->BKP26R,
+    &RTC->BKP27R,
+    &RTC->BKP28R,
+    &RTC->BKP29R,
+    &RTC->BKP30R,
+    &RTC->BKP31R
+};
+
+void
+hal_nvreg_write(unsigned int reg, uint32_t val)
+{
+    if(reg < HAL_NVREG_MAX)
```

Comment:
Please fix the style, a space between `if (` and open braces at the end of same line.